### PR TITLE
[BUG] Fixed Unexpected NaN values in `WindowSummarizer` by modifying `summarize.py` 

### LINKED ISSUE #5781: [BUG] WindowSummarizer unexpected NaN
**Describe the bug**

With Sktime 0.25.0 the WindowSummarizer returns unexpected missing values.

**To Reproduce**

```python
import numpy as np
import pandas as pd
from sktime.transformations.series.summarize import WindowSummarizer

df = pd.DataFrame(np.arange(10))
transformer = WindowSummarizer(lag_feature={"mean": [[-1, 1], [-1, 2]]}, n_jobs=1)
df_transformed = transformer.fit_transform(df)

print(df_transformed)
```

**Expected behavior**

value 0.5 in 1st row 2nd column rather than NaN

**Additional context**

**Versions**
<details>

System:
    python: 3.10.12 | packaged by Anaconda, Inc. | (main, Jul  5 2023, 19:01:18) [MSC v.1916 64 bit (AMD64)]
executable: [c:\Users\ramon.bussing\AppData\Local\anaconda3\envs\enerfore-dev\python.exe](file:///C:/Users/Ram0nB/AppData/Local/anaconda3/envs/enerfore-dev/python.exe)
   machine: Windows-10-10.0.19044-SP0

Python dependencies:
          pip: 23.2.1
       sktime: 0.25.0
      sklearn: 1.3.0
       skbase: 0.6.1
        numpy: 1.22.4
        scipy: 1.8.1
       pandas: 2.1.1
   matplotlib: None
       joblib: 1.3.2
        numba: None
  statsmodels: 0.14.0
     pmdarima: None
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
tensorflow_probability: None

</details>

### PR DESCRIPTION
**Pull request by ShreeshaM07**
#### Reference Issues/PRs

Fixes #5781 

#### What does this implement/fix? Explain your changes.

It ensures that unexpected NaN values are avoided in `WindowSummarizer`

#### Does your contribution introduce a new dependency? If yes, which one?

No

#### Did you add any tests for the change?

added a test in `test_window_summarizer.py` for series

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

**Comment by ShreeshaM07** [comment]
I made a new PR with a better title and details about the tests also added myself as a contributor in `CONTRIBUTORs.md`.
Please let me know if any other modification needs to be made.

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/d
...
       <td align="center" valign="top" width="11.11%"><a href="https://github.com/aaronreidsmith"><img src="https://avatars.githubusercontent.com/u/21350310?v=4?s=100" width="100px;" alt="Aaron Smith"/><br /><sub><b>Aaron Smith</b></sub></a><br /><a href="https://github.com/sktime/sktime/commits?author=aaronreidsmith" title="Code">💻</a></td>
+      <td align="center" valign="top" width="11.11%"><a href="https://github.com/ShreeshaM07"><img src="https://avatars.githubusercontent.com/u/120820143?s=96&v=4" width="100px;" alt="Shreesha M"/><br /><sub><b>Shreesha M</b></sub></a><br /><a href="https://github.com/sktime/sktime/issues?q=author:ShreeshaM07" title="Bug reports">🐛</a> <a href="https://github.com/sktime/sktime/commits?author=ShreeshaM07" title="Documentation">📖</a><a href="https://github.com/sktime/sktime/commits?author=ShreeshaM07" title="Code">💻</a></td>
```
Comment (benHeid) [inline CONTRIBUTORS.md:26]: Please do not update the CONTRIBUTORS.md instead update the .all-contributorsrc. The CONTRIBUTORS.md will be updated automatically.

File: examples/blog_posts/Overview_of_sktime_functionalities_for_forecasting.ipynb
Comment (benHeid) [inline examples/blog_posts/Overview_of_sktime_functionalities_for_forecasting.ipynb:1]: Please revert the changes to that notebook, since they are not part of the issue that is solved by this PR.

File: sktime/transformations/series/tests/test_window_summarizer.py [hunk 2]
```diff
@@ -94,6 +102,13 @@ def count_gt100(x):
 @pytest.mark.parametrize(
     "kwargs, column_names, y, target_cols, truncate",
     [
+        (
```
Comment (benHeid) [inline sktime/transformations/series/tests/test_window_summarizer.py:105]: I would like to see that this test is also executed with a dataFrame, with multi index data and hierarchical data.

File: sktime/transformations/series/tests/test_window_summarizer.py [hunk 3]
```diff
@@ -76,6 +76,14 @@ def count_gt100(x):
         count_gt100: [[3, 2]],
     }
 }
+
+kwargs_series = {
+    "lag_feature": {
+        "mean": [[-1, 1], [-1, 2]],
+    }
+}
+y_series = pd.DataFrame(np.arange(10))
```
Comment (benHeid) [inline sktime/transformations/series/tests/test_window_summarizer.py:85]: You are overwriting y_series. You could probably reuse `y_pd` which is defined previously for the test.

File: sktime/transformations/series/tests/test_window_summarizer.py [hunk 4]
```diff
@@ -76,6 +76,14 @@ def count_gt100(x):
         count_gt100: [[3, 2]],
     }
 }
+
+kwargs_series = {
```
Comment (benHeid) [inline sktime/transformations/series/tests/test_window_summarizer.py:80]: I would propose to rename these kwargs to `kwargs_negative_lags` since they could also be applied to hierarchical data etc.

**Review by benHeid — CHANGES_REQUESTED** [review]
Thank you for adding the test. The remaining comments are just minor improvements. :) 

I think it would be cool if you could also add tests for other data. Just to ensure that this Bug is not reoccurring in other Datatypes.
